packages/squiggle-lang/src/rescript/FunctionRegistry/FunctionRegistry_Helpers.res

```diff
@@ -0,0 +1,121 @@
+open FunctionRegistry_Core
+
+let impossibleError = "Wrong inputs / Logically impossible"
+
+module Wrappers = {
+  let symbolic = r => DistributionTypes.Symbolic(r)
+  let evDistribution = r => ReducerInterface_ExpressionValue.EvDistribution(r)
+  let symbolicEvDistribution = r => r->Symbolic->evDistribution
+}
+
+module Prepare = {
+  type ts = array<frValue>
+  type err = string
+
+  module ToValueArray = {
+    module Record = {
+      let twoArgs = (inputs: ts): result<ts, err> =>
+        switch inputs {
+        | [FRValueRecord([(_, n1), (_, n2)])] => Ok([n1, n2])
+        | _ => Error(impossibleError)
+        }
+    }
+  }
+
+  module ToValueTuple = {
+    let twoDistOrNumber = (values: ts): result<(frValueDistOrNumber, frValueDistOrNumber), err> => {
+      switch values {
+      | [FRValueDistOrNumber(a1), FRValueDistOrNumber(a2)] => Ok(a1, a2)
+      | _ => Error(impossibleError)
+      }
+    }
+
+    module Record = {
+      let twoDistOrNumber = (values: ts): result<(frValueDistOrNumber, frValueDistOrNumber), err> =>
+        values->ToValueArray.Record.twoArgs->E.R.bind(twoDistOrNumber)
+    }
+  }
+}
+
+module Process = {
+  let twoDistsOrNumbersToDist = (
```

Comment:
The error management here is really basic (just uses results with string errors) and could use improvement.